auth-web/src/store/modules/business.ts

```diff
@@ -105,34 +105,35 @@ export default class BusinessModule extends VuexModule {
       })
       .catch(error => {
         console.log('Error getting affiliated entities:', error) // eslint-disable-line no-console
-        return [] as Business[]
+        return [] as []
       })
 
-    // update store with initial results
-    this.setBusinesses(affiliatedEntities)
-
-    // get data for NR entities (not async)
-    const getNrDataPromises = affiliatedEntities.map(entity => {
-      if (entity.corpType.code === CorpTypes.NAME_REQUEST) {
-        return BusinessService.getNrData(entity.businessIdentifier)
+    let affiliatedEntities: Business[] = []
+
+    entityResponse.forEach((resp, i) => {
```

Comment:
is this fast enough? It seems odd to me that your parsing all this data instead of just using the response as is and updating the field names elsewhere in the UI

Reply:
yes, it' not bad. the api takes around 6s to get 2800 data set, the page displays just after api finished. because I use virtual scrolling, it won't render everything.